src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -497,8 +497,8 @@ def _check_value(self, action, value):  # pylint: disable=too-many-statements, t
             else:
                 # `command_source` indicates command values have been parsed, value is an argument
                 parameter = action.option_strings[0] if action.option_strings else action.dest
-                error_msg = "{prog}: '{value}' is not a valid value for '{param}'.".format(
-                    prog=self.prog, value=value, param=parameter)
+                error_msg = "{prog}: '{value}' is not a valid value for '{param}'. Allowed values: {choice}.".format(
+                    prog=self.prog, value=value, param=parameter, choice=', '.join(sorted([str(x) for x in action.choices])))
```

Comment:
Do we need `sorted`? Shall we respect the order in swagger definition?

Reply:
The order in error message is better to be consistent with the order in help info

Reply:
I personally prefer to keep the order because some choices may be more frequently used.